samples/wifi/provisioning/implementation_details.rst

```diff
@@ -0,0 +1,219 @@
+
+   * The `state` field describes the Wi-Fi connection state, whose possible value is defined in `common.proto`.
+   * The `provisioning_info` field described the Wi-Fi provisioning information stored in the device's non-volatile memory (NVM).
+   * The `connection_info` field describes the details of Wi-Fi connection.
+   * The `scan_info` field describes the parameter used for AP scan.
+
+#. The device prepares `Response` message and sends encoded message to the configurator. Depending on the result, one of the response codes defined above is chosen.
+
+`START_SCAN` procedure
+----------------------
+
+1. The configurator prepares `Request` message with ST`ART_SCAN command and sends encoded message to the device.
+#. The device decoded the message and finds the opcode is `START_SCAN`.
+#. The device sends scan request to Wi-Fi subsystem. If scan parameter is provided in the message, it is sent together.
+#. The device prepares `Response` message and sends encoded message to the configurator. Depending on the result, one of the response codes defined above is chosen.
+#. When the scan is done, the device prepares `Result` message and sends encoded message to the configurator.
+
+`STOP_SCAN` procedure
+---------------------
+
+1. The configurator prepares `Request` message with `STOP_SCAN` command and sends encoded message to the device.
+#. The device decoded the message and finds the opcode is `STOP_SCAN`.
+#. The device sends stop scan request to Wi-Fi subsystem.
+#. The device prepares `Response` message and sends encoded message to the configurator. Depending on the result, one of the response codes defined above is chosen.
+
+`SET_CONFIG` procedure
+----------------------
+
+1. The configurator prepares `Request` message with `SET_CONFIG` command and sends encoded message to the device.
+#. The device decoded the message and finds the opcode is `SET_CONFIG`.
+#. The device sends connection request to Wi-Fi subsystem. The connection parameter is sent as well.
+#. The device prepares `Response` message and sends encoded message to the configurator. Depending on the result, one of the response codes defined above is chosen.
+#. When the connection is done, the device prepares `Result` message and sends encoded message to the configurator. The Wi-Fi credential will be saved in flash if connection is successful.
+
+`FORGET_CONFIG` procedure
+-------------------------
+
+1. The configurator prepares `Request` message with `FORGET_CONFIG` command and sends encoded message to the device.
+#. The device decoded the message and finds the opcode is `FORGET_CONFIG`.
+#. The device sends disconnection request to Wi-Fi subsystem, and removes saved Wi-Fi credential if any.
+#. The device prepares `Response` message and sends encoded message to the configurator. Depending on the result, one of the response codes defined above is chosen.
```

Comment:
The rest of these would follow the model I provided above. If you put them in this format, I'll continue looking at the language in the next round.

Reply:
I realize that the procedure is something different to the command. For example, the procedure is to get the device status, and it should use GET_STATUS command, while proper message sequence and content. I will rephrase this part, using unordered list.